index.js

```diff
@@ -320,12 +337,10 @@
   //  xor :: a -> a -> a
   S.xor = R.curry(function(x, y) {
     assertTypeMatch(x, y);
-    switch (true) {
-      case toBoolean(x) !== toBoolean(y): return or(x, y);
-      case R.is(Array, x):                return [];
-      case R.is(Boolean, x):              return false;
-      default:                            return x.empty();
-    }
+    var xBool = toBoolean(x);
+    var yBool = toBoolean(y);
+    var xEmpty = empty(x);
+    return xBool !== yBool ? or(x, y) : xEmpty;
```

Comment:
Looking at this my first reaction is to suggest this change:

```diff
-    var xEmpty = empty(x);
-    return xBool !== yBool ? or(x, y) : xEmpty;
+    return xBool !== yBool ? or(x, y) : empty(x);
```

`empty` has side effects, so this is not equivalent. If `x` does not have an `empty` method we want to throw a descriptive TypeError regardless of whether the empty value is actually needed. Otherwise, `S.xor(S.Left('abc'), S.Right(1))` will evaluate to `Right(1)` but `S.xor(S.Right(1), S.Right(2))` will throw.